cmd/bot/run.go

```diff
@@ -108,15 +108,21 @@ func ProcessUnprovenBotDelegatedWithdrawals(ctx context.Context, log log.Logger,
 				result := db.Model(&unproven).Update("proven", true)
 				if result.Error != nil {
 					log.Error("failed to update proven l2_contract_events", "error", result.Error)
-					return
 				}
 			} else if strings.Contains(err.Error(), "L2OutputOracle: cannot get output for a block that has not been proposed") {
 				// Since the unproven withdrawals are sorted by the on-chain order, we can break here because we know
 				// that the subsequent of the withdrawals are not ready to be proven yet.
 				return
+			} else if strings.Contains(err.Error(), "execution reverted") {
+				// Proven transaction reverted, mark it with the failure reason
+				result := db.Model(&unproven).Update("failure_reason", err.Error())
```

Comment:
Current failure_reason is set to varchar(256), but it cannot be determined that err.Error() will always be within 256 characters. It is recommended to truncate it.

Reply:
Make sense!
Updated based on your suggestion at [c463328](https://github.com/bnb-chain/opbnb-bridge-bot/pull/8/commits/c463328a76a5fc03836f8c42f7b013dd1389de42)

Reply:
Line 161 has the same problem.

Reply:
Oh my fault.
I updated and force-push at [17ef561](https://github.com/bnb-chain/opbnb-bridge-bot/pull/8/commits/17ef561113af38a5e815d99c11163281992784a8).